packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,543 @@
+- Internally we might not need to put in heavy efforts to ensure composable API is very well structured since we will not be exposing it to the consumers. We would use the composable API internally only to derive the final outcome of a compact API.
+- Strong reasons to go with this approach would be the popularity of the compact API within the Table libraries ecosystem
+- Strong reasons to not go with this approach would be the lack of composability and tree-shaking benefits
+
+##### 3. Both Composable & Compact API
+- We could expose both the Composable & Compact API to the consumers
+- This would take the most amount of implementation time and effort since we will have to build both the APIs
+- This will add a lot of complexity to the library and will make it harder to maintain. Any changes to the Table component, we will have to ensure both APIs are updated accordingly and always maintain parity
+- Swapping out the table library in the future would be challenging since any library we choose we will have to ensure that its feasible to build both kinds of APIs
+- Strong reasons to go with this approach would be to give our consumers complete flexibility to choose the API that they are most comfortable with
+- Strong reasons to not go with this approach would be the complexity and maintainability issues as well as education efforts required to educate the consumers about both the APIs
+
+##### Current API Usage Across Razorpay Projects
+> This is an approximate usage count. We identified this by searching through the codebase. Actual numbers may vary slightly.
+ 
+| Project         | Composable API Table Instances | Compact API Table Instances |
+| --------------- | ------------------------------ | --------------------------- |
+| PG Dashboard    | 101                            | 81                          |
+| Admin Dashboard | 19                             | 196                         |
+| X Dashboard     | 6                              | 11                          |
+
+
+##### Blade Team's Recommendation
+**We recommend** going ahead with the **Compact API** since it gives us the following advantages:
+
+- It is the most popular approach within the Table libraries ecosystem which makes it easier for us to swap out the table library in the future
+- It is easier to consume since it is more concise and requires less effort to create a table
+- TypeScript would be able to help us with type checking the API better since it is a single component
+- Learning curve is lesser since its a single component and not a set of components that need to be composed together
+- As per our findings, 70% of the tables at Razorpay are built using the Compact API which would make it relatively easier for us to migrate them to Blade's Table component as well
+
+##### Conclusion
+[To be concluded]
+
+### Final API
```

Comment:
Although API is not yet final, but would be great to have loading and refetch states as well for the table. This have a very good use case when pairing table with react-query style data fetching libraries. Below is a loading state illustration and for refetching a progress bar on top can be indicative of a background refresh.
<img width="851" alt="Screenshot 2023-10-30 at 1 14 29 PM" src="https://github.com/razorpay/blade/assets/95009160/5ecffb86-c529-48a2-aff0-50722e15f1cb">

Reply:
Loading state is already a part of the API. Do we have any such use-cases right now where we have progress bar on top or refetch mechanism in our products?

Reply:
@anilprzp @chaitanyadeorukhkar folks will this be a use case?

https://github.com/razorpay/blade/assets/24494390/8cd34b40-a723-413c-9c1b-4b2c0a72ff55

Reply:
I will forward this to design and check.